File: temp_audit/analyze_web.py
import os
from playwright.sync_api import sync_playwright

def run_analysis():
    with sync_playwright() as p:
        # Launch browser
        browser = p.chromium.launch(headless=True)
        context = browser.new_context()
        page = context.new_page()

        # Track console logs
        logs = []
        page.on("console", lambda msg: logs.append(f"[{msg.type}] {msg.text}"))
        page.on("pageerror", lambda err: logs.append(f"[ERROR] {err.message}"))

        # Navigate to the app
        print("Navigating to http://localhost:5173...")
        try:
            page.goto("http://localhost:5173", wait_until="networkidle")
        except Exception as e:
            print(f"Failed to navigate: {e}")
            browser.close()
            return

        # Create output directory for screenshots
        os.makedirs("temp_audit/screenshots", exist_ok=True)

        # Audit viewports
        viewports = [
            {"name": "desktop", "width": 1920, "height": 1080},
            {"name": "tablet", "width": 768, "height": 1024},
            {"name": "mobile", "width": 375, "height": 667}
        ]

        for vp in viewports:
            print(f"Capturing {vp['name']} screenshot...")
            page.set_viewport_size({"width": vp["width"], "height": vp["height"]})
            # Wait a bit for animations/resize to settle
            page.wait_for_timeout(1000)
            page.screenshot(path=f"temp_audit/screenshots/{vp['name']}.png", full_page=True)

        # Save logs
        with open("temp_audit/audit_logs.txt", "w", encoding="utf-8") as f:
            f.write("\n".join(logs))

        # Check for specific elements (optional but good for a "pro" audit)
        print("Performing element audit...")
        all_images = page.locator("img").all()
        missing_alt = []
        for img in all_images:
            alt = img.get_attribute("alt")
            if not alt or alt.strip() == "":
                missing_alt.append(img.get_attribute("src") or "unknown source")
        
        with open("temp_audit/element_audit.txt", "w", encoding="utf-8") as f:
            f.write(f"Images missing alt text: {len(missing_alt)}\n")
            for src in missing_alt:
                f.write(f"- {src}\n")

        print("Analysis complete.")
        browser.close()

if __name__ == "__main__":
    run_analysis()

File: temp_audit/analyze_web.js
import { chromium } from 'playwright';
import fs from 'fs';
import path from 'path';

const APP_URL = 'http://localhost:8080';
const OUTPUT_DIR = 'temp_audit';
const SCREENSHOTS_DIR = path.join(OUTPUT_DIR, 'screenshots');

async function runAnalysis() {
    if (!fs.existsSync(SCREENSHOTS_DIR)) {
        fs.mkdirSync(SCREENSHOTS_DIR, { recursive: true });
    }

    const browser = await chromium.launch({ headless: true });
    const context = await browser.newContext();
    const page = await context.newPage();

    const logs = [];
    page.on('console', msg => logs.push(`[${msg.type()}] ${msg.text()}`));
    page.on('pageerror', err => logs.push(`[ERROR] ${err.message}`));

    console.log(`Navigating to ${APP_URL}...`);
    try {
        await page.goto(APP_URL, { waitUntil: 'networkidle' });
    } catch (e) {
        console.error(`Failed to navigate: ${e.message}`);
        await browser.close();
        return;
    }

    const viewports = [
        { name: 'desktop', width: 1920, height: 1080 },
        { name: 'tablet', width: 768, height: 1024 },
        { name: 'mobile', width: 375, height: 667 }
    ];

    for (const vp of viewports) {
        console.log(`Capturing ${vp.name} screenshot...`);
        await page.setViewportSize({ width: vp.width, height: vp.height });
        await page.waitForTimeout(1000);
        await page.screenshot({ path: path.join(SCREENSHOTS_DIR, `${vp.name}.png`), fullPage: true });
    }

    fs.writeFileSync(path.join(OUTPUT_DIR, 'audit_logs.txt'), logs.join('\n'));

    console.log('Performing element audit...');
    const images = await page.locator('img').all();
    const missingAlt = [];
    for (const img of images) {
        const alt = await img.getAttribute('alt');
        const src = await img.getAttribute('src');
        if (!alt || alt.trim() === '') {
            missingAlt.push(src || 'unknown source');
        }
    }

    const report = `Images missing alt text: ${missingAlt.length}\n${missingAlt.map(src => `- ${src}`).join('\n')}`;
    fs.writeFileSync(path.join(OUTPUT_DIR, 'element_audit.txt'), report);

    console.log('Analysis complete.');
    await browser.close();
}

runAnalysis().catch(err => {
    console.error('Audit failed:', err);
    process.exit(1);
});
